[Break XPU][Inductor] Generalize device-bias code and fix test_graph_partition for XPU

etaf opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #147727
* __->__ #148178
* #148155

This PR generalized the device-bias code introduced by #147038 . And align the behavior between XPU and CUDA on add + mm + pointwise pattern (for XPU, from addmm + pointwise to mm + fused_add_pointwise) , which fix the failed test case `test_graph_partiton` on XPU. 

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

etaf committed:
[Break XPU][Inductor] Generalize device-bias code and fix test_graph_partition for
XPU.

[ghstack-poisoned]

alexbaden commented on test/inductor/test_torchinductor.py:
```diff
@@ -14021,7 +14021,7 @@ def fn(x):
 
             fn_c = torch.compile(fn)
-            x = torch.rand(16, device="cuda")
+            x = torch.rand(16, device=GPU_TYPE)
```
out of curiosity, why do we prefer `GPU_TYPE` or `self.device` in different places in this file?

etaf replied:
This is to follow the original design. For example, some cases are meant to test scenarios where both CPU and CUDA tensors coexist. If we replace "cuda" with self.device, it would not align with the original design of the case. Changing it to GPU_TYPE ensures consistency with the original design.

etaf committed:
Update on "[Break XPU][Inductor] Generalize device-bias code and fix test_graph_partition for XPU"


This PR generalized the device-bias code introduced by #147038 . And align the behavior between XPU and CUDA on add + mm + pointwise pattern (for XPU, from addmm + pointwise to mm + fused_add_pointwise) , which fix the failed test case `test_graph_partiton` on XPU. 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

etaf commented:
@pytorchbot label “keep-going”

etaf commented:
@pytorchbot label keep-going

etaf committed:
Update on "[Break XPU][Inductor] Generalize device-bias code and fix test_graph_partition for XPU"


This PR generalized the device-bias code introduced by #147038 . And align the behavior between XPU and CUDA on add + mm + pointwise pattern (for XPU, from addmm + pointwise to mm + fused_add_pointwise) , which fix the failed test case `test_graph_partiton` on XPU. 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

etaf commented:
@pytorchbot merge
